internal/services/redisenterprise/redis_enterprise_database_resource.go

```diff
@@ -255,45 +258,50 @@ func resourceRedisEnterpriseDatabaseRead(d *pluginsdk.ResourceData, meta interfa
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.RedisEnterpriseDatabaseID(d.Id())
+	id, err := databases.ParseDatabaseID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.RedisEnterpriseName, id.DatabaseName)
+	resp, err := client.Get(ctx, *id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			log.Printf("[INFO] Redis Enterprise Database %q does not exist - removing from state", d.Id())
 			d.SetId("")
 			return nil
 		}
-		return fmt.Errorf("retrieving Redis Enterprise Database %q (Resource Group %q / Cluster Name %q): %+v", id.DatabaseName, id.ResourceGroup, id.RedisEnterpriseName, err)
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
 	}
 
-	keysResp, err := client.ListKeys(ctx, id.ResourceGroup, id.RedisEnterpriseName, id.DatabaseName)
+	keysResp, err := client.ListKeys(ctx, *id)
 	if err != nil {
-		return fmt.Errorf("listing keys for Redis Enterprise Database %q (Resource Group %q / Cluster Name %q): %+v", id.DatabaseName, id.ResourceGroup, id.RedisEnterpriseName, err)
+		return fmt.Errorf("listing keys for %s: %+v", *id, err)
 	}
 
 	d.Set("name", id.DatabaseName)
-	d.Set("resource_group_name", id.ResourceGroup)
-	d.Set("cluster_id", parse.NewRedisEnterpriseClusterID(id.SubscriptionId, id.ResourceGroup, id.RedisEnterpriseName).ID())
-
-	if props := resp.DatabaseProperties; props != nil {
-		d.Set("client_protocol", props.ClientProtocol)
-		d.Set("clustering_policy", props.ClusteringPolicy)
-		d.Set("eviction_policy", props.EvictionPolicy)
-		if err := d.Set("module", flattenArmDatabaseModuleArray(props.Modules)); err != nil {
-			return fmt.Errorf("setting `module`: %+v", err)
+	d.Set("resource_group_name", id.ResourceGroupName)
+	clusterId := redisenterprise.NewRedisEnterpriseID(id.SubscriptionId, id.ResourceGroupName, id.ClusterName)
+	d.Set("cluster_id", clusterId.ID())
+
+	if model := resp.Model; model != nil {
+		if props := model.Properties; props != nil {
+			d.Set("client_protocol", props.ClientProtocol)
+			d.Set("clustering_policy", props.ClusteringPolicy)
+			d.Set("eviction_policy", props.EvictionPolicy)
+			if err := d.Set("module", flattenArmDatabaseModuleArray(props.Modules)); err != nil {
+				return fmt.Errorf("setting `module`: %+v", err)
+			}
+			// if err := d.Set("persistence", flattenArmDatabasePersistence(props.Persistence)); err != nil {
+			// 	return fmt.Errorf("setting `persistence`: %+v", err)
+			// }
```

Comment:
how come this is commented out?

Reply:
(as discussed offline) this is coming from the original data source - I think this is WIP from @WodansSon?